pycbc/io/hdf.py

```diff
@@ -217,6 +217,60 @@ def save(self, outname):
             f['segments/%s/end' % key] = self.seg[key]['end'][:]
         f.close()
 
+class MultiifoStatmapData(DictArray):
```

Comment:
This should subclass StatmapData to avoid duplicate code (unless the plan is to delete the original class .. but then I'd still advocate subclassing for now and cleaning up when removing the old code).

Reply:
I will look at it during the weekend, to do it as a base class of StatmapData, won't have time before saturday.

Reply:
@spxiwh  I changed this to be a subclass of StatmapData, but I could not use the constructor of that class, because it hardcodes the groups, It allowed me to remove that save method which remained identical, but the rest of the functions had to be changed. I used the super class, with StatmapData inside so it uses the constructor of the Dictarray class, but codeclimate is complaining about it now. Which version should be used?

Reply:
@tapaimarton @spxiwh - @GarethDaviesGW and I think we can address this subclassing and reduce the total amount of code by making 'groups' into a kwarg with a default value.  However it seems messy to try to combine our changes with Marton's in this pull request .. in fact I'm not sure even how to do this.  So either we should merge this now and then make a new request for the codeclimate fixes, or start an entire new request .. thoughts?

Reply:
Whatever is easiest to you. You *can* checkout Marton's branch, make changes there which you submit to a local branch in your checkout and then open a pull request back onto Marton's branch (which will then appear here).

But I emphasize again: Do whatever's easiest to you.

Reply:
If you want me to force-merge this so you can proceed, I will do that.

Reply:
FWIW I agree with force-merging so long as nothing is affected for standard use and other users, which it doesn't look like it is

Reply:
I'm not sure if having a request to pull onto Marton's branch would require Marton to merge it.  (.. then we don't know if he is available to do that at the moment.)

Reply:
That would require Marton to merge it, yes.